spec/appraisal/matrix/extensions/appraisal_file_spec.rb

```diff
@@ -9,26 +9,78 @@
     subject { appraisal_matrix(**desired_gems) }
 
     context "with a maximum version specified" do
-      let(:desired_gems) { { rails: { min: "6.1", max: "7.0" } } }
+      let(:desired_gems) { { rails: { min: "6.1", max: "7.1" } } }
 
-      it "is not implemented yet" do
-        expect { subject }.to raise_error("TODO: Version request options not implemented yet")
+      before do
+        expect(Appraisal::Matrix::AppraiseFileWithMatrix::VersionArray).to receive(:new).with(min: "6.1", max: "7.1").and_wrap_original do |original_method, *args, **kwargs|
+          original_method.call(*args, **kwargs).tap do |version_array|
+            expect(version_array).to receive(:versions_to_test).with(:rails, Gem::Version.new("6.1"), Gem::Version.new("7.1"), :minor).and_return(["6.1", "7.0"])
+          end
+        end
```

Comment:
I feel like this is a sign that we should change some code to make this easier to mock.

Perhaps creating a `RubyGemsClient` class that we can request versions for a gem for.

Reply:
That's an interesting heuristic, I'll try some things out!